### PR TITLE
GA Migration Java PostCommit DataFlow v1 and v2

### PR DESCRIPTION
**Pull request by fernando-wizeline**
As part of the migration of Precommit and Postcommit Jobs from Jenkins to GA in self-hosted runners, this PR contains:

Migrated workflow [job-postcommit-java-dataflow-v1.yml](https://github.com/fernando-wizeline/beam/blob/gam-postcommit-java-dataflow-v1-v2/.github/workflows/job-postcommit-java-dataflow-v1.yml)
Migrated workflow [job-postcommit-java-dataflow-v2.yml](https://github.com/fernando-wizeline/beam/blob/gam-postcommit-java-dataflow-v1-v2/.github/workflows/job-postcommit-java-dataflow-v2.yml)

The migrated workflows were removed from [.test-infra/jenkins/README.md](https://github.com/fernando-wizeline/beam/blob/gam-postcommit-java-dataflow-v1-v2/.test-infra/jenkins/README.md) and added to [CI.md](https://github.com/fernando-wizeline/beam/blob/gam-postcommit-java-dataflow-v1-v2/CI.md)

DO NOT MERGE until the effort to use self-hosted runners is completed https://github.com/apache/beam/pull/22703

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job-postcommit-java-dataflow-v1.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    name: Run PostCommit Java Dataflow v1
+    runs-on: [self-hosted, ubuntu-20.04]
+    strategy:
+      fail-fast: false
+      matrix: #JDK11 and 17 compilation will fail due to changes in ByteBuffer class since JDK9
+        #TODO: Retest once issue https://github.com/apache/beam/issues/23068 gets resolved.
+        java: ${{fromJson(needs.set-properties.outputs.properties).JavaTestProperties.SUPPORTED_VERSIONS}}
+        exclude:
+          - java: "11"
+          - java: "17"
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-dataflow-v1.yml:58]: As mentioned in https://github.com/apache/beam/pull/23077#discussion_r988104603 we shouldn't be removing the Jenkins files until this exclusion is no longer needed

File: .github/workflows/job-postcommit-java-dataflow-v2.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    needs: set-properties
+    name: Run PostCommit Java Dataflow v2
+    runs-on: [self-hosted, ubuntu-20.04]
+    strategy:
+      fail-fast: false
+      matrix: #JDK11 and 17 compilation will fail due to changes in ByteBuffer class since JDK9
+        #TODO: Retest once issue https://github.com/apache/beam/issues/23068 gets resolved.
+        java: ${{fromJson(needs.set-properties.outputs.properties).JavaTestProperties.SUPPORTED_VERSIONS}}
+        exclude:
+          - java: "11"
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-dataflow-v2.yml:57]: Same comment for this jenkins file

File: .github/workflows/job-postcommit-java-dataflow-v1.yml (hunk 1 above)
Comment (fernando-wizeline) [reply .github/workflows/job-postcommit-java-dataflow-v1.yml:58]: Thanks for the heads up, Danny!
Added jenkins files back

**Review by damccorm — APPROVED** [review]
LGTM, thanks

**Pull request closed without merging**